app/views/form/page.html.erb

```diff
@@ -17,7 +19,11 @@
       </div>
     <% end %>
 
-    <%= f.hidden_field :previous_page, value: page_key %>
+    <% if @case_log.soft_errors.present? && @case_log.soft_errors.keys.first == page_info["soft_validations"]&.keys&.first %>
```

Comment:
It might be worth speaking to Jon about how we want the back link to behave in certain scenarios. My gut feeling is that it should behave like the browser back button. It might also be worth seeing what other gov services do for back button behaviour

Reply:
Reverted hijacking the back link and made the form page not cache instead. Still sub-optimal but slightly less so...